task_sdk/src/airflow/sdk/definitions/dag.py

```diff
@@ -769,10 +766,12 @@ def partial_subset(
         memo = {id(self.task_dict): None, id(self.task_group): None}
         dag = copy.deepcopy(self, memo)  # type: ignore
 
-        if isinstance(task_ids_or_regex, (str, Pattern)):
-            matched_tasks = [t for t in self.tasks if re2.findall(task_ids_or_regex, t.task_id)]
+        if isinstance(task_ids, str):
+            matched_tasks = [t for t in self.tasks if str(task_ids) in t.task_id]
         else:
-            matched_tasks = [t for t in self.tasks if t.task_id in task_ids_or_regex]
+            matched_tasks = [
+                t for t in self.tasks if any(task_id for task_id in task_ids if str(task_id) in t.task_id)
+            ]
```

Comment:
What was there before seemed easier to understand to me, and was working for `Iterable[str]`.

Reply:
I had doubts about changing this. I first thought we were missing a case. After checking again, they are the same. I agree, converted back. It is easier and simpler.